src/test/java/com/internship/backend/model/NewDateTest.java

```diff
@@ -0,0 +1,203 @@
+package com.internship.backend.model;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class NewDateTest {
+
+    @Test
+    void testEquals() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,12,5,3,2024);
+        NewDate newDate3 = new NewDate(0,14,20,2,2024);
+
+        assertEquals(newDate1, newDate2);
+        assertNotEquals(newDate1, newDate3);
+    }
+
+    @Test
+    void isAfter() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,10,5,3,2024);
+        NewDate newDate3 = new NewDate(0,14,20,2,2024);
+
+        assertTrue(newDate1.isAfter(newDate2));
+        assertTrue(newDate2.isAfter(newDate3));
+        assertFalse(newDate3.isAfter(newDate2));
+    }
+
+    @Test
+    void isBefore() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,10,5,3,2024);
+        NewDate newDate3 = new NewDate(0,14,20,2,2024);
+
+        assertTrue(newDate3.isBefore(newDate2));
+        assertTrue(newDate3.isBefore(newDate1));
+        assertTrue(newDate2.isBefore(newDate1));
+        assertFalse(newDate1.isBefore(newDate2));
+    }
+
+    @Test
+    void testSameDateComparison(){
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+
+        assertFalse(newDate1.isAfter(newDate1));
+        assertFalse(newDate1.isBefore(newDate1));
+    }
+
+    @Test
+    void getMinute() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,10,5,3,2024);
+        NewDate newDate3 = new NewDate(0,14,20,2,2024);
+
+        assertEquals(newDate1.getMinute(), 30);
+        assertEquals(newDate2.getMinute(), 30);
+        assertEquals(newDate3.getMinute(), 0);
+        assertNotEquals(newDate1.getMinute(), 12);
+    }
+
+    @Test
+    void getHour() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,10,5,3,2024);
+        NewDate newDate3 = new NewDate(0,14,20,2,2024);
+
+        assertEquals(newDate1.getHour(), 12);
+        assertEquals(newDate2.getHour(), 10);
+        assertEquals(newDate3.getHour(), 14);
+        assertNotEquals(newDate1.getHour(), 15);
+    }
+
+    @Test
+    void getDay() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
+        NewDate newDate2 = new NewDate(30,10,5,3,2024);
+        NewDate newDate3 = new NewDate(0,14,20,2,2024);
+
+        assertEquals(newDate1.getDay(), 5);
+        assertEquals(newDate2.getDay(), 5);
+        assertEquals(newDate3.getDay(), 20);
+        assertNotEquals(newDate1.getDay(), 15);
+    }
+
+    @Test
+    void getMonth() {
+
+        NewDate newDate1 = new NewDate(30,12,5,3,2024);
```

Comment:
you could have directly used a LocalDateTime java object instead of creating a separate object (NewDate) for representing time and date